### ISSUE TITLE
getObject returns random object

### ISSUE THREAD

**Issue by mishantrop** [modxcms/revolution#13215]
Random category has been assigned to element while creating (click on plus icon in elements tree).
(e.g. manager/controllers/default/element/chunk/create.class.php, method getCategory).

### Step to reproduce
```
$category = $modx->getObject('modCategory', 0);
if (is_object($category)) {
	echo $category->get('id').'<br/>';
	echo $category->get('category');
} else {
	echo 'type: '.gettype($category);
}
```

### Observed behavior
```
17
Blah-blah
```

### Expected behavior
$category must be NULL?

### Environment
MODX Revolution 2.5.2-pl, PHP 7

**Comment by OptimusCrime** [comment]
@mishantrop Why did you close this issue? I have experienced something like this myself.

**Comment by mishantrop** [comment]
@OptimusCrime because it is duplicate of issue #13187 and pull-request #13207

Also I fixed it in one file of my project:
manager/controllers/default/element/chunk/create.class.php
```
public function getCategory(array $scriptProperties = array()) {
	/* grab default category if specified */
	if (isset($scriptProperties['category'])) {
		$this->category = $this->modx->getObject('modCategory', ['id' => (int)$scriptProperties['category']]);
	} else {
		$this->category = null;
	}
	return $this->category;
}
```
But there are many files which uses this bug.

**Comment by OptimusCrime** [comment]
That makes sense, my PR fixes that issue, but I have no idea just why it happened in the first place. I don't know why the getObject failed in the first place.

However, it is always smart to tag duplicates if that happens to be the case. You are right that it is the same problem, but this issue goes more into the underlying code.

**Comment by spectrum48k** [comment]
May be similar to my problem:

in case of no cache files created for db objects of 'someClassKey':
->getObject('someClassKey'); and ->getCollection('someClassKey') (or ->getMany('someAliasOfClassKey')) after that
shoots to cache file with same md5 prefix and different nesting of arrays inside cache files. This causes recurse and nesting level error when collection loads from cache. So ->getObject() without unique criteria is bad practice for current xPDO caching principle. I'l try to create issue with this info

**Comment by opengeek** [comment]
This was an issue created by a security fix in 2.5.2 via changes to xPDO's getObject behavior. This has been reverted for 2.5.3 and should now be resolved in all instances.
